eclipse/src/saros/ui/wizards/AddProjectToSessionWizard.java

```diff
@@ -668,4 +687,106 @@ private void displaySaveDirtyEditorsDialog(final List<IEditorPart> dirtyEditors)
       for (IEditorPart editor : dirtyEditors) editor.doSave(new NullProgressMonitor());
     }
   }
+
+  private void storeCurrentProjectNameMapping(final JID jid) {
+    Map<String, String> currentProjectNameMapping = new HashMap<>();
+
+    for (final ProjectNegotiationData data : negotiation.getProjectNegotiationData()) {
+      final String projectID = data.getProjectID();
+      currentProjectNameMapping.put(
+          data.getProjectName(), namePage.getTargetProjectName(projectID));
+    }
+
+    mappingStorage.updateMapping(jid, currentProjectNameMapping);
+  }
+
+  private static final class ResourceMappingStorage {
+    private static final String RESOURCE_MAPPING_STORE_PREFIX = "resource.mapping.storage.";
+
+    private static final CharSequence DELIMITER = ":";
+
+    private static final CharSequence ESCAPE_CHARACTER = "\\";
+
+    private static final CharSequence ESCAPE_SEQUENCE =
+        ESCAPE_CHARACTER.toString() + DELIMITER.toString();
+
+    private static final Pattern SPLIT_PATTERN =
+        Pattern.compile(
```

Comment:
I am not very well versed when it comes to regular expressions. Did you test that it actually does what it is supposed to?

Reply:
Well it actually worked even without escaping at all. I have to test this in code as I cannot create directories containing ":" under Windows.

Reply:
Before store: {Hello=World, 1:2:3:4:5:6:7:8:9:0=::::::::::::::::::::::::::::::, Foo:Bar=Foo:Foo}
Stored string: Foo\:Bar:Foo\:Foo:Hello:World:1\:2\:3\:4\:5\:6\:7\:8\:9\:0:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:\:
Read from store: {Hello=World, 1:2:3:4:5:6:7:8:9:0=::::::::::::::::::::::::::::::, Foo:Bar=Foo:Foo}
Write == Read ? true